arroyo/processing/strategies/streaming/transform.py

```diff
@@ -225,12 +231,40 @@ def parallel_transform_worker_initializer(
         custom_initialize_func()
 
 
+@dataclass
+class ParallelTransformResult(Generic[TTransformed]):
+    """
+    Results of applying the transform function to a batch
+    of messages in parallel.
+
+    `index_processed_til` holds the index of the next
+    message to be processed from the input batch. If
+    something goes wrong while processing the batch and
+    that message needs to be reprocessed, the index can
+    be passed back to the function to start processing
+    from there.
+
+    `output_batch` contains all successfully processed
+    messages.
+
+    `bad_messages` contains `InvalidMessage` exceptions
+    caught during processing. These messages (exceptions)
+    should be passed to a DLQ by raising the
+    `InvalidBatchedMessages` exception.
+    """
+
+    index_processed_til: int
+    output_batch: MessageBatch[TTransformed]
+    bad_messages: MutableSequence[InvalidMessage]
+
+
 def parallel_transform_worker_apply(
```

Comment:
Please have a unit test for this function with the changes you added.

Reply:
I've already added a test here which tests the case of some bad and some good messages:
https://github.com/getsentry/arroyo/blob/2535e2f7391e35493e04e6c14045b2e0ebb0c737/tests/processing/strategies/test_streaming.py#L417

Is there another case you'd like tested?

Also see my comment:
https://github.com/getsentry/arroyo/pull/61#discussion_r854545804
I'm uncertain if this ok^